code/src/main/kotlin/com/expediagroup/sdk/core/logging/common/LoggerDecorator.kt

```diff
@@ -0,0 +1,35 @@
+package com.expediagroup.sdk.core.logging.common
+
+import org.slf4j.Logger
+
+class LoggerDecorator(private val logger: Logger) : Logger by logger {
+    override fun info(msg: String) = logger.info(decorate(msg))
+
+    override fun warn(msg: String) = logger.warn(decorate(msg))
+
+    override fun debug(msg: String) = logger.debug(decorate(msg))
+
+    override fun error(msg: String) = logger.error(decorate(msg))
+
+    override fun trace(msg: String) = logger.trace(decorate(msg))
+
+    fun info(msg: String, vararg tags: String) = logger.info(decorate(msg, tags.toSet()))
+
+    fun warn(msg: String, vararg tags: String) = logger.warn(decorate(msg, tags.toSet()))
+
+    fun debug(msg: String, vararg tags: String) = logger.debug(decorate(msg, tags.toSet()))
+
+    fun error(msg: String, vararg tags: String) = logger.error(decorate(msg, tags.toSet()))
+
+    fun trace(msg: String, vararg tags: String) = logger.trace(decorate(msg, tags.toSet()))
+
+    private fun decorate(msg: String, tags: Set<String>? = null): String = buildString {
+        append("[${Constant.EXPEDIA_GROUP_SDK}] - ")
+        tags?.let {
+            append("[")
+            append(it.joinToString(", "))
+            append("] - ")
+        }
+        append(msg.trim())
+    }
```

Comment:
Do we expect the clients to add a tag for the specific sdk implementation? Should we give a specific required field for that so we know which sdk is producing the log?

Reply:
Tags are like metadata holders in the logs, not really something we have to rely on to identify the SDK. For SDK identification, we are using `UserAgent` header that should hold all data about the SDK (name, version, namespace), but I don't mind adding a tag to represent the SDK name :)

This is a sample of the logs at the moment, 
<img width="1827" alt="Screenshot 2024-12-03 at 14 19 10" src="https://github.com/user-attachments/assets/16af9d7b-5521-428b-9bf8-04122c65372d">

Anyway, our logging mechanism is based on one interceptor at the moment, we will need to have more granular control over this and be able to log from any place. @mohnoor94 has a ticket to spike how should we do that, and if we need the logging interceptor at all.